source/server/admin/stats_handler.cc

```diff
@@ -82,17 +82,26 @@ Http::Code StatsHandler::handlerStats(absl::string_view url,
     return Http::Code::BadRequest;
   }
 
+  const Stats::CustomStatNamespaces& custom_stat_namespaces = server_.api().customStatNamespaces();
```

Comment:
perf nit: WDYT of adding an `empty()` predicate to CustomStatNamespaces and have a faster path when there are no prefixes?

There's a bit of overhead, it looks to me, in attempting to strip every stat when there's nothing to do if no custom namespaces are registered.

I think this would also apply to the existing Prometheus, though that has a lot of per-stat munging anyway.

Reply:
this one -- this will add per-counter overhead without benefit (for non-wasm users) and there can be a lot of counters. It's not difficult, right?

Reply:
yeah this should be easy - OK I will do in my working hours tomorrow:)

Reply:
done; https://github.com/envoyproxy/envoy/pull/18127/commits/8c00c5f22a14f0822b8c36a3e2ce938b4318c86b